extensions-contrib/aliyun-oss-extensions/src/main/java/org/apache/druid/storage/aliyun/OssStorageDruidModule.java

```diff
@@ -103,4 +105,11 @@ public OSS initializeOssClient(OssClientConfig inputSourceConfig)
   {
     return inputSourceConfig.buildClient();
   }
+
+  @Provides
+  @LazySingleton
+  public Supplier<OSS> initializeOssClientSupplier(OssClientConfig inputSourceConfig)
+  {
+    return Suppliers.memoize(inputSourceConfig::buildClient);
```

Comment:
nit: I think leaving a comment here would help people understand that this is lazily supplied to avoid needing to set unnecessary properties